platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -322,6 +176,75 @@ class MGLStyleValueTransformer {
+        // defaultValue => default
+        if (styleFunction.defaultValue) {
+            NSCAssert([styleFunction.defaultValue isKindOfClass:[MGLStyleConstantValue class]], @"Default value must be constant");
+            rawFunction[@"default"] = [(MGLStyleConstantValue<ObjCType> *)styleFunction.defaultValue rawValue];
+        }
+        
+        // interpolationBase => base
+        if (styleFunction.interpolationBase) {
+            // TODO: does this direct cast work?
+            rawFunction[@"base"] = (NSNumber *) styleFunction.interpolationBase;
+        }
+        
+        // stops
+        if ([styleFunction isKindOfClass:[MGLCameraStyleFunction class]]) {
+            // zoom-only function
+            __block NSMutableArray *stops = [[NSMutableArray alloc] init];
+            [styleFunction.stops enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull zoomKey, MGLStyleConstantValue<ObjCType> * _Nonnull outputValue, BOOL * _Nonnull stop) {
+                NSArray *rawStop = @[zoomKey, [outputValue rawValue]];
+                [stops addObject:rawStop];
+            }];
+        } else if ([styleFunction isKindOfClass:[MGLSourceStyleFunction class]]) {
+            rawFunction[@"property"] = ((MGLSourceStyleFunction<ObjCType> *)styleFunction).attributeName;
+            // property-only function
+            __block NSMutableArray *stops = [[NSMutableArray alloc] init];
+            [styleFunction.stops enumerateKeysAndObjectsUsingBlock:^(NSObject * _Nonnull propertyKey, MGLStyleConstantValue<ObjCType> * _Nonnull outputValue, BOOL * _Nonnull stop) {
+                NSArray *rawStop = @[propertyKey, [outputValue rawValue]];
+                [stops addObject:rawStop];
+            }];
+        } else if ([styleFunction isKindOfClass:[MGLCompositeStyleFunction class]]) {
+            // zoom-and-property function
+            MGLCompositeStyleFunction<ObjCType> *compositeStyleFunction = (MGLCompositeStyleFunction<ObjCType> *)styleFunction;
+            rawFunction[@"property"] = compositeStyleFunction.attributeName;
+            
+            __block NSMutableArray *stops = [[NSMutableArray alloc] init];
+            [compositeStyleFunction.stops enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull zoomKey, NSDictionary * _Nonnull stopValue, BOOL * _Nonnull stop) {
+                for (NSObject *valueKey in stopValue.allKeys) {
+                    NSDictionary *stopKey = @{
+                         @"zoom": zoomKey,
+                         @"value": valueKey
+                     };
+                    MGLStyleConstantValue<ObjCType> *outputValue = stopValue[valueKey];
+                    NSCAssert([outputValue isKindOfClass:[MGLStyleConstantValue<ObjCType> class]], @"Stop outputs should be MGLStyleConstantValues");
+                    NSArray *rawStop = [NSArray arrayWithObjects:stopKey, [outputValue rawValue]];
+                    [stops addObject:rawStop]; // TODO weird: why doesn't stops.addObject(rawStop) work?
```

Comment:
In Objective-C, the [syntax for calling a method](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithObjects/WorkingwithObjects.html) is `[object sendMessage:message withArgument:argument options:options]`. The syntactic sugar for an instance method taking no parameters is `dot.syntax`, which is meant for getting a `@property` (since the getter of a property is an instance method taking no parameters).

Incidentally, since a property’s getter is a method like any other, you should be aware that getting a property may sometimes be an expensive operation with side effects. [Some developers](https://www.bignerdranch.com/blog/dot-notation-syntax/) advise against using dot notation to emphasize this point. In fact, there’s no downside to using bracket notation for methods and properties alike, other than forcing your poor code reviewer to read Lisp.